crates/hc/src/save.rs

```diff
@@ -96,3 +107,65 @@ pub fn list(hc_dir: PathBuf, verbose: usize) -> anyhow::Result<()> {
     msg!("{}", out);
     Ok(())
 }
+
+lazy_static::lazy_static! {
+    static ref FILE_LOCKS: tokio::sync::Mutex<Vec<usize>> = tokio::sync::Mutex::new(Vec::new());
+}
+
+/// Lock this setup as running live and advertise the port
+pub async fn lock_live(mut hc_dir: PathBuf, path: &Path, port: u16) -> anyhow::Result<()> {
+    use std::io::Write;
+    std::fs::create_dir_all(&hc_dir)?;
+    let paths = load(hc_dir.clone())?;
+    let index = match paths.into_iter().enumerate().find(|p| p.1 == path) {
+        Some((i, _)) => i,
+        None => return Ok(()),
+    };
+    hc_dir.push(format!(".hc_live_{}", index));
+    match std::fs::OpenOptions::new()
+        .write(true)
+        .create_new(true)
+        .open(hc_dir)
+    {
+        Ok(mut file) => {
+            writeln!(file, "{}", port)?;
+            let mut lock = FILE_LOCKS.lock().await;
+            lock.push(index);
+        }
+        Err(e) => match e.kind() {
+            std::io::ErrorKind::AlreadyExists => {}
+            _ => return Err(e.into()),
+        },
+    }
+
+    Ok(())
+}
+
+/// Load setup ports
```

Comment:
```suggestion
/// For each registered setup, if it has a lockfile, return the port of the running conductor,
/// otherwise return None. 
/// The resulting Vec has the same number of elements as lines in the `.hc` file
```